### ISSUE TITLE
evaluate vs. __call__

### ISSUE THREAD

**Issue by joergfunger** [BAMresearch/BayesianInference#20]
The inference problem currently only has a __call_function that returns a dictionary (for the experiment) of vectors (concatenation of all sensor outputs). I think it would be nice to have an interface that allows to propagate the complete model error to the output, thus a function that returns a dictionary (experiment) of dictionaries (sensors) - which I could implement. Just asking myself, if we still need the __call__ function (so that I would just change __call__) and then have another helper function on the top level of the inference problem (so not implementing an evaluate and __call__ in parallel in all classes like ModelError and ForwardModel). I think in these classes a single __call__ returning the dict of dict version would be sufficient, and then we provide an evaluate_as_vector function in the inferenceProblem.

**Comment by joergfunger** [comment]
I think I misinterpreted the current implementation, the call function already returns a dict of OrderedDict.
